cpp/test/sg/logger.cpp

```diff
@@ -36,4 +37,34 @@ TEST(Logger, Test) {
   ASSERT_TRUE(Logger::get().shouldLogFor(CUML_LEVEL_WARN));
 }
 
+std::string logged = "";
+void exampleCallback(int lvl, const char* msg) { logged = std::string(msg); }
+
+TEST(Logger, callback) {
+  Logger::get().setCallback(exampleCallback);
```

Comment:
Do we need a cleanup somewhere so this callback doesn't run here for the remainder of our tests?

Reply:
Whoops! Good catch. Fixing that one right away...

Reply:
Fixed in latest version.